internal narrowing conversion error in ppu/include/vectormath/cpp/quat_aos.h

Potential issue with PSL1GHT:

Reproduction code: `make -f Makefile.ps3` from https://github.com/emukidid/wii64-ps3 current master (commit 85b23eeaf0c3ef224cdde6074bcda99339b6274f for future reference)

Error: 
```
/usr/local/ps3dev/ppu/include/vectormath/cpp/quat_aos.h:502:100: error: narrowing conversion of '2147483648' from 'unsigned int' to 'int' inside { } [-Wnarrowing]
     return Quat( vec_xor( quat.get128(), ((vec_float4)(vec_int4){0x80000000,0x80000000,0x80000000,0}) ) );```

I've got a similar problem trying to build the `rsxtest` sample from PSL1GHT ( https://github.com/ps3dev/PSL1GHT/tree/master/samples/graphics/rsxtest )

by running `make`, I get the same error about the narrowing conversion.

Following @miigotu comment, I tried casting to `vec_uint4` (modified line 502 in `/usr/local/ps3dev/ppu/include/vectormath/cpp/quat_aos.h` )

Tried to compile again, and to my surprise, I've got a very weird error (at least for my limited knowledge):
```
% make
main.cpp
/Users/dparrino/github/psl1ght-libs/samples/psl1ght/graphics/rsxtest/source/main.cpp: In function 'int main(int, const char**)':
/Users/dparrino/github/psl1ght-libs/samples/psl1ght/graphics/rsxtest/source/main.cpp:556:6: warning: variable 'ret' set but not used [-Wunused-but-set-variable]
  s32 ret,i;
      ^~~
/Users/dparrino/github/psl1ght-libs/samples/psl1ght/graphics/rsxtest/source/main.cpp: In function 'void program_exit_callback()':
/Users/dparrino/github/psl1ght-libs/samples/psl1ght/graphics/rsxtest/source/main.cpp:410:1: internal compiler error: in rs6000_savres_routine_name, at config/rs6000/rs6000.c:28661
 }
 ^
libbacktrace could not find executable to open
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://gcc.gnu.org/bugs/> for instructions.
make[1]: *** [main.o] Error 1
make: *** [build] Error 2
```

did I truly broke gcc, or I did something wrong on my side?

thanks for the help

--- a/common/vectormath/ppu/cpp/quat_aos.h
+++ b/common/vectormath/ppu/cpp/quat_aos.h
@@ -499,7 +499,7 @@ inline const Vector3 rotate( Quat quat, Vector3 vec )
 
 inline const Quat conj( Quat quat )
 {
-    return Quat( vec_xor( quat.get128(), ((vec_float4)(vec_int4){0x80000000,0x80000000,0x80000000,0}) ) );
+    return Quat( vec_xor( quat.get128(), ((vec_float4)(vec_uint4){0x80000000,0x80000000,0x80000000,0}) ) );
 }
 
 inline const Quat select( Quat quat0, Quat quat1, bool select1 )

--- a/common/vectormath/spu/cpp/quat_aos.h
+++ b/common/vectormath/spu/cpp/quat_aos.h
@@ -385,7 +385,7 @@ inline const Vector3 rotate( Quat quat, Vector3 vec )
 
 inline const Quat conj( Quat quat )
 {
-    return Quat( spu_xor( quat.get128(), ((vec_float4)(vec_int4){0x80000000,0x80000000,0x80000000,0}) ) );
+    return Quat( spu_xor( quat.get128(), ((vec_float4)(vec_uint4){0x80000000,0x80000000,0x80000000,0}) ) );
 }
 
 inline const Quat select( Quat quat0, Quat quat1, bool select1 )

--- a/samples/graphics/rsxtest/Makefile
+++ b/samples/graphics/rsxtest/Makefile
@@ -30,7 +30,7 @@ CONTENTID	:=	UP0001-$(APPID)_00-0000000000000000
 # options for code generation
 #---------------------------------------------------------------------------------
 
-CFLAGS		=	-Os -Wall -mcpu=cell $(MACHDEP) $(INCLUDE)
+CFLAGS		=	-Wall -mcpu=cell $(MACHDEP) $(INCLUDE)
 CXXFLAGS	=	$(CFLAGS)
 
 LDFLAGS		=	$(MACHDEP) -Wl,-Map,$(notdir $@).map

--- a/samples/graphics/rsxtest_spu/Makefile
+++ b/samples/graphics/rsxtest_spu/Makefile
@@ -30,7 +30,7 @@ CONTENTID	:=	UP0001-$(APPID)_00-0000000000000000
 # options for code generation
 #---------------------------------------------------------------------------------
 
-CFLAGS		=	-Os -Wall -mcpu=cell $(MACHDEP) $(INCLUDE)
+CFLAGS		=	-Wall -mcpu=cell $(MACHDEP) $(INCLUDE)
 CXXFLAGS	=	$(CFLAGS)
 
 LDFLAGS		=	$(MACHDEP) -Wl,-Map,$(notdir $@).map

